compiler/rustc_pattern_analysis/src/pat.rs

```diff
@@ -192,7 +192,7 @@ impl<'p, Cx: PatCx> PatOrWild<'p, Cx> {
     }
     pub(crate) fn ctor(self) -> &'p Constructor<Cx> {
         match self {
-            PatOrWild::Wild => &Wildcard,
+            PatOrWild::Wild => const { &Wildcard },
```

Comment:
Not even the compiler builds without some changes.^^

What happens here is that by making the const expression `&Wildcard`, the lifetime extension rules kick in, and those still do value-based reasoning and therefore allow this constant without ever involving promotion.